### PR TITLE
Force workflow deletion from redis based on environment config.

### PR DESCRIPTION
**Pull request by manan164**
Workflow will be deleted forcefully if environment property ```force.workflow.deletion``` is true. If it is false then ```workflowStatusListenerEnabled``` will be used to decide on removal strategy.

### CODE REVIEW

File: core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java [hunk 1]
```diff
@@ -662,7 +662,12 @@ void completeWorkflow(Workflow wf) {
         queueDAO.remove(DECIDER_QUEUE, workflow.getWorkflowId());    //remove from the sweep queue
 
         //Remove workflow from redis irrespective of status listener.
-        workflowStatusListener.onWorkflowCompleted(workflow);
+        if(config.getForceWorkflowDeletionFromRedis()) {
```
Comment (naveenchlsn) [inline core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java]: Consider writing the condition as 
if( config.getForceWorkflowDeletionFromRedis() || workflow.getWorkflowDefinition().isWorkflowStatusListenerEnabled()) {
workflowStatusListener.onWorkflowCompleted(workflow);
}

Comment (manan164) [reply core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java]: I thought of it earlier but this is for better understanding.

Comment (naveenchlsn) [reply core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java]: We are duplicating the code here. We can have the if..else if you think this is more intuitive. 
But please extract out the code from both methods to a single method. 
https://refactoring.guru/smells/duplicate-code

Comment (manan164) [reply core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java]: did changes.

**Pull request merged**